### DISCUSSION TITLE
How to use $t('text') in <script> block?

### DISCUSSION THREAD

**Discussion by HendrikJan** [nuxt/nuxt#17656] [Questions]
I've installed '@nuxtjs/i18n' successfully.

Now I can use translations in my `<template>` block:
```html
<div>{{ $t(translateMe) }}</div>
```
but I cannot find how to use translations in my code block:
```javascript
const translated = ref( $t(translateMe) ); // <-- this fails because $t is undefined
```

Does anybody know how to use translations with `@nuxt/i18n` within the `<script>` tag?

**Comment by TheAlexLichter** [comment]
~~The i18n module is only available for Nuxt2 at the moment (according to [modules.nuxtjs.org](https://modules.nuxtjs.org/?q=i18n)).~~

For Bridge and Nuxt3, the `next` package is [on the way](https://github.com/nuxt-community/i18n-module/blob/next/playground/bridge/pages/index.vue). See the linked example for using it in Nuxt Bridge ☺️

**Reply by HendrikJan** [reply]
That would be good enough as I am using nuxt-bridge.
(discussing it here because nuxt-bridge repo seems to have no discussions enabled)

**Comment by mnlj** [comment]
You need to use the [next version of @nuxtjs/i18n](https://github.com/nuxt-community/i18n-module/tree/next) and then within the `<script setup>` tag you can use 
```
const { t } = useI18n();
const translated = ref(t(translateMe));
```

**Reply by HendrikJan** [reply]
This might work in some setups.

In my setup (with Nuxt-bridge) I get "useI18n is not defined"
I'll just find ways to put the translations in the `<template>` for now.

**Reply by TheAlexLichter** [reply]
Feel free to check out the [example](https://github.com/nuxt-community/i18n-module/blob/next/playground/bridge/pages/index.vue) for bridge then ☺️ 

```vue
<script setup lang="ts">
import { useI18n } from 'https://github.com/mashpie/i18n-node'
const { t, locale, locales, localeProperties } = useI18n()
const text = computed(() => t('hello'))
</script>
```
